src/core/ctx.ts

```diff
@@ -19,7 +19,8 @@ export function createContext(options: Options = {}, root = process.cwd()) {
     dts: preferDTS = isPackageExists('typescript'),
   } = options
 
-  const dirs = options.dirs?.map(dir => resolve(root, dir))
+  const filePattern = '*.{tsx,jsx,ts,js,mjs,cjs,mts,cts}'
+  const dirs = options.dirs?.map(dir => resolve(root, dir, !dir.includes('*') ? filePattern : ''))
```

Comment:
This does not seem to handle `!` right? Can we extract this as a function and do some unit test?

Reply:
yes, unplugin-vue-components already has pr: https://github.com/antfu/unplugin-vue-components/pull/645

if that pr passed, I will change  unplugin-auto-import code

Reply:
I'm afraid this pr won't pass, so i am waiting.